src/providers/postgres/qgspostgresexpressioncompiler.cpp

```diff
@@ -32,3 +36,161 @@ QString QgsPostgresExpressionCompiler::quotedValue( const QVariant& value, bool&
   return QgsPostgresConn::quotedValue( value );
 }
 
+static const QMap<QString, QString>& functionNamesSqlFunctionsMap()
+{
+  static QMap<QString, QString> fnNames;
+  if ( fnNames.isEmpty() )
+  {
+    // math functions
+    fnNames.insert( QStringLiteral( "sqrt" ), QStringLiteral( "sqrt" ) );
+    fnNames.insert( QStringLiteral( "radians" ), QStringLiteral( "radians" ) );
+    fnNames.insert( QStringLiteral( "degrees" ), QStringLiteral( "degrees" ) );
+    fnNames.insert( QStringLiteral( "abs" ), QStringLiteral( "abs" ) );
+    fnNames.insert( QStringLiteral( "cos" ), QStringLiteral( "cos" ) );
+    fnNames.insert( QStringLiteral( "sin" ), QStringLiteral( "sin" ) );
+    fnNames.insert( QStringLiteral( "tan" ), QStringLiteral( "tan" ) );
+    fnNames.insert( QStringLiteral( "acos" ), QStringLiteral( "acos" ) );
+    fnNames.insert( QStringLiteral( "asin" ), QStringLiteral( "asin" ) );
+    fnNames.insert( QStringLiteral( "atan" ), QStringLiteral( "atan" ) );
+    fnNames.insert( QStringLiteral( "atan2" ), QStringLiteral( "atan2" ) );
+    fnNames.insert( QStringLiteral( "exp" ), QStringLiteral( "exp" ) );
+    fnNames.insert( QStringLiteral( "ln" ), QStringLiteral( "ln" ) );
+    fnNames.insert( QStringLiteral( "log" ), QStringLiteral( "log" ) );
+    fnNames.insert( QStringLiteral( "log10" ), QStringLiteral( "log" ) );
+    fnNames.insert( QStringLiteral( "round" ), QStringLiteral( "round" ) );
+    fnNames.insert( QStringLiteral( "floor" ), QStringLiteral( "floor" ) );
+    fnNames.insert( QStringLiteral( "ceil" ), QStringLiteral( "ceil" ) );
+    fnNames.insert( QStringLiteral( "pi" ), QStringLiteral( "pi" ) );
+    // geometry functions
+    fnNames.insert( QStringLiteral( "azimuth" ), QStringLiteral( "ST_Azimuth" ) );
+    fnNames.insert( QStringLiteral( "x" ), QStringLiteral( "ST_X" ) );
```

Comment:
We can't compile x/y as the QGIS versions have some differences. Eg qgis one will return centroid x for a polygon input.